src/js/utils/browser.js

```diff
@@ -5,196 +5,198 @@
-  let version = result && parseFloat(result[1]);
-
-  if (!version && (/Trident\/7.0/i).test(USER_AGENT) && (/rv:11.0/).test(USER_AGENT)) {
-    // IE 11 has a different user agent string than other IE versions
-    version = 11.0;
-  }
-
-  return version;
-}());
+export let CHROME_VERSION = null;
 
 /**
  * Whether or not this is desktop Safari.
  *
  * @static
- * @const
  * @type {Boolean}
  */
-export const IS_SAFARI = (/Safari/i).test(USER_AGENT) && !IS_CHROME && !IS_ANDROID && !IS_EDGE;
+export let IS_SAFARI = false;
 
 /**
  * Whether or not this is a Windows machine.
  *
  * @static
- * @const
  * @type {Boolean}
  */
-export const IS_WINDOWS = (/Windows/i).test(USER_AGENT);
+export let IS_WINDOWS = false;
 
 /**
- * Whether or not this device is touch-enabled.
+ * Whether or not this device is an iPad.
  *
  * @static
- * @const
  * @type {Boolean}
  */
-export const TOUCH_ENABLED = Boolean(Dom.isReal() && (
-  'ontouchstart' in window ||
-  window.navigator.maxTouchPoints ||
-  window.DocumentTouch && window.document instanceof window.DocumentTouch));
+export let IS_IPAD = false;
 
 /**
- * Whether or not this device is an iPad.
+ * Whether or not this device is an iPhone.
  *
  * @static
- * @const
  * @type {Boolean}
  */
-export const IS_IPAD = (/iPad/i).test(USER_AGENT) ||
-  (IS_SAFARI && TOUCH_ENABLED && !(/iPhone/i).test(USER_AGENT));
+// The Facebook app's UIWebView identifies as both an iPhone and iPad, so
+// to identify iPhones, we need to exclude iPads.
+// http://artsy.github.io/blog/2012/10/18/the-perils-of-ios-user-agent-sniffing/
+export let IS_IPHONE = false;
 
 /**
- * Whether or not this device is an iPhone.
+ * Whether or not this device is touch-enabled.
  *
  * @static
  * @const
  * @type {Boolean}
  */
-// The Facebook app's UIWebView identifies as both an iPhone and iPad, so
-// to identify iPhones, we need to exclude iPads.
-// http://artsy.github.io/blog/2012/10/18/the-perils-of-ios-user-agent-sniffing/
-export const IS_IPHONE = (/iPhone/i).test(USER_AGENT) && !IS_IPAD;
+export const TOUCH_ENABLED = Boolean(Dom.isReal() && (
+  'ontouchstart' in window ||
+  window.navigator.maxTouchPoints ||
+  window.DocumentTouch && window.document instanceof window.DocumentTouch));
+
+const UAD = window.navigator && window.navigator.userAgentData;
+
+if (UAD) {
+  // If userAgentData is present, use it instead of userAgent to avoid warnings
+  // Currently only implemented on Chromium
+  // userAgentData does not expose Android version, so ANDROID_VERSION remains `null`
+
+  IS_ANDROID = UAD.platform === 'Android';
+  IS_CHROME = Boolean(UAD.brands.find(b => b.brand === 'Google Chrome'));
+  IS_EDGE = Boolean(UAD.brands.find(b => b.brand === 'Microsoft Edge'));
+  CHROME_VERSION = (UAD.brands.find(b => b.brand === 'Chromium') || {}).version || null;
+  IS_WINDOWS = UAD.platform === 'Windows';
+}
+
+if (!CHROME_VERSION) {
+  // no userAgentData, or a browser that has added userAgentData since this was added
```

Comment:
i.e. if Firefox adds `userAgentData`, it might specify its brand as `Firefox` or `Mozilla Firefox` so we can't really add in advance.